brooklyn-server/server-cli/src/main/java/org/apache/brooklyn/cli/ItemLister.java

```diff
@@ -98,88 +115,150 @@ public Void call() throws Exception {
+                for (URL url: urls) {
+                    List<Map<?,?>> entities = new ArrayList<>();
+                    List<Map<?,?>> policies = new ArrayList<>();
+                    List<Map<?,?>> enrichers = new ArrayList<>();
+                    List<Map<?,?>> locations = new ArrayList<>();
+                    List<Map<?,?>> locationResolvers = new ArrayList<>();
+
+                    String yamlContent = Files.toString(new File(url.getFile()), Charset.forName("UTF-8"));
+
+                    catalog.addItems(yamlContent);
+                    for (CatalogItem item: catalog.getCatalogItems()) {
+                        AbstractBrooklynObjectSpec<?,?> spec = catalog.createSpec(item);
+                        Map<String,Object> itemDescriptor = ItemDescriptors.toItemDescriptor((Class<? extends BrooklynObject>) spec.getType(), false);
+                        List<EntityConfigSummary> config = new ArrayList<>();
+
+                        if (!itemDescriptor.containsKey("name")) {
+                            itemDescriptor.put("name", item.getDisplayName());
+                        }
+                        if (!itemDescriptor.containsKey("description")) {
+                            itemDescriptor.put("description", item.getDescription());
+                        }
+                        if (!itemDescriptor.containsKey("iconUrl")) {
+                            itemDescriptor.put("iconUrl", item.getIconUrl());
+                        }
+                        for (SpecParameter<?> param: spec.getParameters()) {
+                            EntityConfigSummary entityConfigSummary = EntityTransformer.entityConfigSummary(param.getType(),
+                                    param.getLabel(), null, MutableMap.<String,URI>of());
+                            config.add(entityConfigSummary);
+                        }
+                        itemDescriptor.put("config", config);
+                        if (item.getCatalogItemType().name().equals("ENTITY")) {
+                            entities.add(itemDescriptor);
+                        } else if (item.getCatalogItemType().name().equals("POLICY")) {
+                            policies.add(itemDescriptor);
+                        } else if (item.getCatalogItemType().name().equals("LOCATION")) {
+                            locations.add(itemDescriptor);
+                        }
+                    }
+                    result = ImmutableMap.<String, Object>builder()
+                            .put("entities", entities)
+                            .put("policies", policies)
+                            .put("enrichers", enrichers)
+                            .put("locations", locations)
+                            .put("locationResolvers", locationResolvers)
+                            .build();
+
+                    jsonList.add(toJson(result));
                 }
```

Comment:
Remove the catalog items after processing them, so the next `addItems` call works with a clean catalog - avoiding conflicts.